packages/ingest/test/test-queue.js

```diff
@@ -312,3 +312,107 @@ test.serial('enqueueParsePdrMessage uses the executionNamePrefix if specified',
   t.true(executionName.startsWith(executionNamePrefix));
   t.true(executionName.length > executionNamePrefix.length);
 });
+
+test.serial('enqueueParsePdrMessage does not overwrite the collection or provider with additionalCustomMeta if specified', async (t) => {
+  const {
+    queueUrl,
+    templateBucket,
+    stackName,
+    workflow,
+  } = t.context;
+
+  const collection = {
+    name: 'ABC123',
+    version: '001',
+  };
+
+  const provider = {
+    id: 'prov1',
+  };
+
+  await queue.enqueueParsePdrMessage({
+    collection,
+    parentExecutionArn: randomId(),
+    parsePdrWorkflow: workflow,
+    pdr: {},
+    provider,
+    stack: stackName,
+    systemBucket: templateBucket,
+    queueUrl,
+    additionalCustomMeta: {
+      collection: {
+        name: 'XYZ789',
+        version: '999',
+      },
+      provider: {
+        id: 'PROV99',
+      },
+    },
+  });
+
+  const receiveMessageResponse = await sqs().receiveMessage({
+    QueueUrl: queueUrl,
+    MaxNumberOfMessages: 10,
+    WaitTimeSeconds: 1,
+  }).promise();
+
+  const actualMessage = JSON.parse(receiveMessageResponse.Messages[0].Body);
+
+  const messageCollection = actualMessage.meta.collection;
+  const messageProvider = actualMessage.meta.provider;
+
+  t.deepEqual(collection, messageCollection);
+  t.deepEqual(provider, messageProvider);
+});
+
+test.serial('enqueueGranuleIngestMessage does not overwrite the collection or provider with additionalCustomMeta if specified', async (t) => {
+  const {
+    queueUrl,
+    templateBucket,
+    stackName,
+    workflow,
+  } = t.context;
+
+  const collection = {
+    name: 'ABC123',
+    version: '001',
+  };
+
+  const provider = {
+    id: 'prov1',
+  };
+
+  await queue.enqueueGranuleIngestMessage({
+    collection,
+    parentExecutionArn: randomId(),
+    granuleIngestWorkflow: workflow,
+    pdr: {},
+    provider,
+    stack: stackName,
+    systemBucket: templateBucket,
+    queueUrl,
+    additionalCustomMeta: {
+      collection: {
+        name: 'XYZ789',
+        version: '999',
+      },
+      provider: {
+        id: 'PROV99',
+      },
+    },
+  });
+
+  const receiveMessageResponse = await sqs().receiveMessage({
+    QueueUrl: queueUrl,
+    MaxNumberOfMessages: 10,
+    WaitTimeSeconds: 1,
+  }).promise();
+
+  const actualMessage = JSON.parse(receiveMessageResponse.Messages[0].Body);
+
+  const messageCollection = actualMessage.meta.collection;
+  const messageProvider = actualMessage.meta.provider;
+
+  t.deepEqual(collection, messageCollection);
+  t.deepEqual(provider, messageProvider);
+});
```

Comment:
I have a question that is probably due to my lack of context and understanding, but why do we need to test that `enqueueParsePdrMessage` and `enqueueGranuleIngestMessage` wouldn't overwrite `collection` and `provider` if those are possibly defined in `additionalCustomMeta`? It's not clear to me that there would be a scenario where it would be overridden so I am wondering why this needs to be tested.

Reply:
This came out [a discussion between me and mark](https://github.com/nasa/cumulus/pull/2054#discussion_r566382511) about whether we should give users the capability to override `collection` and `provider` in the `meta` using this new functionality. We settled on no, so I changed the ordering of the object decomposition and added these tests to ensure that is maintained in the future.